ts/store/reducers/entities/services/servicePreference.ts

```diff
@@ -0,0 +1,62 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import { getType } from "typesafe-actions";
+import { createSelector } from "reselect";
+import { fromNullable } from "fp-ts/lib/Option";
+import { Action } from "../../../actions/types";
+import {
+  loadServicePreference,
+  upsertServicePreference
+} from "../../../actions/services/servicePreference";
+import { ServicePreferenceResponse } from "../../../../types/services/ServicePreferenceResponse";
+import { getNetworkErrorMessage } from "../../../../utils/errors";
+import { IndexedById } from "../../../helpers/indexer";
+import { toError, toLoading, toSome, toUpdating } from "../../IndexedByIdPot";
+import { GlobalState } from "../../types";
+
+export type ServicePreferenceState = IndexedById<
+  pot.Pot<ServicePreferenceResponse, Error>
+>;
+
+const servicePreferenceReducer = (
+  state: ServicePreferenceState = {},
+  action: Action
+): ServicePreferenceState => {
+  switch (action.type) {
+    case getType(loadServicePreference.request):
+      return toLoading(action.payload, state);
+    case getType(upsertServicePreference.request):
+      return toUpdating(action.payload.id, state, {
+        id: action.payload.id,
+        kind: "success",
+        value: {
+          inbox: action.payload.inbox,
+          notifications: action.payload.notifications,
+          email: action.payload.email
+        }
```

Comment:
```suggestion
        value: action.payload
```

Reply:
Payload actually contains `id` too, i preferred to avoid redundant information